test/components/connect.spec.js

```diff
@@ -118,13 +257,238 @@ describe('React', () => {
+        <Provider store={store}>
+          {() => (
+            <Container />
+          )}
+        </Provider>
+      );
+
+      const connector = TestUtils.findRenderedComponentWithType(tree, Container);
+      expect(spy.calls.length).toBe(0);
+      connector.componentWillUnmount();
+      expect(spy.calls.length).toBe(1);
+    });
+
+    it('should shallowly compare the selected state to prevent unnecessary updates', () => {
+      const store = createStore(stringBuilder);
+      const spy = expect.createSpy(() => ({}));
+      function render({ string }) {
+        spy();
+        return <div string={string}/>;
+      }
+
+      @connect(
+        state => ({string: state}),
+        dispatch => ({ dispatch })
+        )
+      class Container extends Component {
+        render() {
+          return render(this.props);
+        }
+      }
+
+      const tree = TestUtils.renderIntoDocument(
+        <Provider store={store}>
+          {() => (
+            <Container />
+          )}
+        </Provider>
+      );
+
+      const div = TestUtils.findRenderedDOMComponentWithTag(tree, 'div');
+      expect(spy.calls.length).toBe(1);
+      expect(div.props.string).toBe('');
+      store.dispatch({ type: 'APPEND', body: 'a'});
+      expect(spy.calls.length).toBe(2);
+      store.dispatch({ type: 'APPEND', body: 'b'});
+      expect(spy.calls.length).toBe(3);
+      store.dispatch({ type: 'APPEND', body: ''});
+      expect(spy.calls.length).toBe(3);
+    });
+
+    it('should recompute the state slice when the select prop changes', () => {
+      const store = createStore(combineReducers({
+        a: () => 42,
+        b: () => 72
+      }));
+
+      function selectA(state) {
+        return { result: state.a };
+      }
+
+      function selectB(state) {
+        return { result: state.b };
+      }
+
+      function render({ result }) {
+        return <div>{result}</div>;
+      }
+
+      function getContainer(select) {
+        return (
+          @connect(select)
+          class Container extends Component {
+            render() {
+              return this.props.children(this.props);
+            }
+          }
+        );
+      }
+
+      class OuterContainer extends Component {
+        constructor() {
+          super();
+          this.state = { select: selectA };
+        }
+
+        render() {
+          return (
+            <Provider store={store}>
+              {() => {
+                const Container = getContainer(this.state.select);
+                return (
+                  <Container>
+                    {render}
+                  </Container>
+                );
+              }}
+            </Provider>
+          );
+        }
+      }
+
+      let tree = TestUtils.renderIntoDocument(<OuterContainer />);
+      let div = TestUtils.findRenderedDOMComponentWithTag(tree, 'div');
+      expect(div.props.children).toBe(42);
+
+      tree.setState({ select: selectB });
```

Comment:
I don't think this test makes sense anymore. `select` isn't a prop now, you must specify it during the class definition.
I'm not sure *why* it breaks but you can remove it.